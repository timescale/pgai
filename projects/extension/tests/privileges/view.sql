\set users {bob,fred,alice,jill}

-- check view privileges
\! rm -f view.actual
select
  n.nspname as "schema"
, k.relname as "view"
, u as "user"
, p as "privilege"
, case has_table_privilege(u, k.oid, p) when true then 'YES' else 'no' end as granted
from unnest(:'users'::text[]) u
inner join pg_namespace n on (n.nspname = any(array['ai', 'wiki']))
inner join pg_class k on (n.oid = k.relnamespace and k.relkind in ('v'))
cross join unnest(array['select']) p
order by n.nspname, k.relname, u, p
\g (format=aligned) view.actual
