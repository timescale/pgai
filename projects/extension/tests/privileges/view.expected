 schema |        view        | user  | privilege | granted 
--------+--------------------+-------+-----------+---------
 ai     | secret_permissions | alice | select    | YES
 ai     | secret_permissions | bob   | select    | no
 ai     | secret_permissions | fred  | select    | no
 ai     | secret_permissions | jill  | select    | YES
 ai     | vectorizer_status  | alice | select    | YES
 ai     | vectorizer_status  | bob   | select    | no
 ai     | vectorizer_status  | fred  | select    | no
 ai     | vectorizer_status  | jill  | select    | YES
 wiki   | post_embedding     | alice | select    | YES
 wiki   | post_embedding     | bob   | select    | no
 wiki   | post_embedding     | fred  | select    | YES
 wiki   | post_embedding     | jill  | select    | YES
(12 rows)

